Add ArchuUnit generation

MSaguer opened the pull request.

MSaguer commented on src/main/resources/generator/server/javatool/arch/HexagonalArchTest.java.mustache:
```diff
@@ -0,0 +1,226 @@
+  private static Collection<String> getPackageAnnotatedBy(Class<? extends Annotation> annotationClass) throws AssertionError {
+    try {
+      return Files
+        .walk(Paths.get("src", "main", "java", "tech", "jhipster", "beer"))
```
need to be genererated

DanielFran commented on src/main/java/tech/jhipster/lite/generator/server/javatool/arch/domain/ArchUnit.java:
```diff
@@ -0,0 +1,27 @@
+
+public class ArchUnit {
+
+  private static final String ARCHUNIT_VERSION = "0.22.0";
+
```
```suggestion
```

DanielFran commented on src/main/java/tech/jhipster/lite/generator/server/javatool/arch/domain/ArchUnit.java:
```diff
@@ -0,0 +1,27 @@
+      .build();
+  }
+
+  public static String getArchUnitVersion() {
+    return ARCHUNIT_VERSION;
+  }
+
```
```suggestion
```

DanielFran commented on src/main/java/tech/jhipster/lite/generator/server/javatool/arch/domain/JavaArchUnitDomainService.java:
```diff
@@ -0,0 +1,52 @@
+
+  @Override
+  public void addArchUnitMavenPlugin(Project project) {
+    buildToolService.addProperty(project, ArchUnit.getPropertyName(), ArchUnit.getArchUnitVersion());
```
```suggestion
    buildToolService
      .getVersion(project, "archunit-junit5")
      .ifPresentOrElse(
        version -> buildToolService.addProperty(project, ArchUnit.getPropertyName(), version),
        () -> {
          throw new GeneratorException("Version not found: archunit-junit5");
        }
      );
```

DanielFran commented:
Please add the version to the pom.xml: https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/dependencies/pom.xml#L33

MSaguer commented:
Thanks @DanielFran for the review.
I've moved the version configuration to the pom.xml file and also added the logback configuration as requested in #584

pascalgrimaud commented on src/main/java/tech/jhipster/lite/generator/server/javatool/arch/domain/JavaArchUnitDomainService.java:
```diff
@@ -0,0 +1,80 @@
+
+    projectRepository.template(project, SOURCE, "BusinessContext.java", getPath(MAIN_JAVA, packageNamePath));
+    projectRepository.template(project, SOURCE, "SharedKernel.java", getPath(MAIN_JAVA, packageNamePath));
+    projectRepository.template(project, SOURCE, "HexagonalArchTest.java", getPath(TEST_JAVA, packageNamePath, TECH_INFRA_PRIMARY_PATH));
```
the destination is not correct, should not be in primary

pascalgrimaud commented on src/main/java/tech/jhipster/lite/generator/server/javatool/arch/domain/JavaArchUnitDomainService.java:
```diff
@@ -0,0 +1,80 @@
+public class JavaArchUnitDomainService implements JavaArchUnitService {
+
+  public static final String SOURCE = "server/javatool/arch";
+  public static final String TECH_INFRA_PRIMARY_PATH = "technical/infrastructure/primary";
```
no need this destination

pascalgrimaud commented on src/test/java/tech/jhipster/lite/generator/server/javatool/arch/application/JavaArchUnitAssertFiles.java:
```diff
@@ -0,0 +1,52 @@
+import tech.jhipster.lite.generator.project.domain.Project;
+
+public class JavaArchUnitAssertFiles {
+
```
plz, add a private constructor

pascalgrimaud commented on tests-ci/generate.sh:
```diff
@@ -24,6 +24,7 @@ springboot() {
   callApi "/api/build-tools/maven"
   callApi "/api/servers/java/base"
   callApi "/api/servers/java/jacoco-minimum-coverage"
+  callApi "/api/servers/java/arch"
```
should not be by default when generating springboot application, you could move it under another config, like full-default

pascalgrimaud commented:
Thanks for your work @MSaguer !

MSaguer commented:
Hi @pascalgrimaud
I've created the open collective expense https://opencollective.com/generator-jhipster/expenses/66173

Better late than never ;)

pascalgrimaud commented:
@MSaguer : approved
